libraries/core/lib/objective/BranchObjectiveFunction.ts

```diff
@@ -89,7 +116,8 @@ export class BranchObjectiveFunction<
     const branchDistance = this.branchDistance.calculate(
       closestCoveredBranchTrace.condition_ast,
       closestCoveredBranchTrace.condition,
-      closestCoveredBranchTrace.variables
+      closestCoveredBranchTrace.variables,
+      trueOrFalse
```

Comment:
I assume this is to restore the opposite branch calculation without having to select the smallest value?

Reply:
Well, I came to the conclusion that selecting the smallest value does not work because if you cover the true branch (i.e. distance = 0) then the false branch always gives a non-zero branch distance. So if you take the smallest value you always end up with 0.

This allows us to calculate for a specific branch.

Reply:
So, my previous assumption that this logic should be checked turned out to be true